File: Backend-python/app/services/user_service.py
from app.repositories.user_repository import find_by_id, update_profile as repo_update_profile
from app.services.auth_service import hash_password

def update_profile(data, file):

    id_usuario = data.get("id_usuario")
    nombre = data.get("nombre_completo")
    password_actual = data.get("password_actual")

    if not id_usuario or not password_actual:
        raise Exception("Datos incompletos")

    user = find_by_id(id_usuario)

    if not user:
        raise Exception("Usuario no encontrado")

    hashed = hash_password(password_actual)

    if user[3] != hashed:
        raise Exception("Contraseña actual incorrecta")

    foto_key = f"Fotos_Perfil/{file.filename}" if file else user[4]

    # 👇 AQUÍ ESTABA EL ERROR
    repo_update_profile(id_usuario, nombre or user[2], foto_key)

    return {
        "success": True,
        "message": "Perfil actualizado correctamente"
    }

File: Backend-Nodejs/src/services/user.service.js
const crypto = require("crypto");
const userRepository = require("../repositories/user.repository");

function hashPassword(password) {
    return crypto.createHash("md5").update(password).digest("hex");
}

exports.updateProfile = async (data, file) => {

    const { id_usuario, nombre_completo, password_actual } = data;

    if (!id_usuario || !password_actual) {
        throw new Error("Datos incompletos");
    }

    const user = await userRepository.findById(id_usuario);

    if (!user) {
        throw new Error("Usuario no encontrado");
    }

    const hashed = hashPassword(password_actual);

    if (user.password !== hashed) {
        throw new Error("Contraseña actual incorrecta");
    }

    const fotoKey = file ? `Fotos_Perfil/${file.originalname}` : user.foto_perfil;

    await userRepository.updateProfile(
        id_usuario,
        nombre_completo || user.nombre_completo,
        fotoKey,
        user.password 
    );
    return {
        success: true,
        message: "Perfil actualizado correctamente"
    };
};